packages/labs/analyzer/src/lib/javascript/jsdoc.ts

```diff
@@ -15,120 +26,264 @@ import {AnalyzerInterface, NamedJSDocInfo, NodeJSDocInfo} from '../model.js';
  */
 
 /**
- * Remove line feeds from JSDoc summaries, so they are normalized to
+ * Returns true if given node has a JSDoc tag
+ */
+export const hasJSDocTag = (node: ts.Node, tag: string) => {
+  return ts.getJSDocTags(node).some((t) => t.tagName.text === tag);
+};
+
+/**
+ * Remove line feeds from JSDoc comments, so they are normalized to
  * unix `\n` line endings.
  */
-const normalizeLineEndings = (s: string) => s.replace(/\r/g, '');
+const normalizeLineEndings = (s: string) => s.replace(/\r/g, '').trim();
 
-// Regex for parsing name, summary, and descriptions from JSDoc comments
-const parseNameDescSummaryRE =
-  /^\s*(?<name>[^\s:]+)([\s-:]+)?(?<summary>[^\n\r]+)?([\n\r]+(?<description>[\s\S]*))?$/m;
+// Regex for parsing name, type, and description from JSDoc comments
+const parseNameTypeDescRE =
+  /^(?<name>\S+)(?:\s+{(?<type>.*)})?(?:\s+-\s+)?(?<description>[\s\S]*)$/m;
 
-// Regex for parsing summary and description from JSDoc comments
-const parseDescSummaryRE =
-  /^\s*(?<summary>[^\n\r]+)\r?\n\r?\n(?<description>[\s\S]*)$/m;
+// Regex for parsing name and description from JSDoc comments
+const parseNameDescRE = /^(?<name>^\S+)(?:\s?-\s+)?(?<description>[\s\S]*)$/m;
+
+// Regex for parsing optional name and description from JSDoc comments, where
+// the dash is required before the description (syntax for `@slot` tag, whose
+// default slot has no name)
+const parseNameDashDescRE =
+  /^(?<name>^\S*)?(?:\s+-\s+(?<description>[\s\S]*))?$/m;
+
+const getJSDocTagComment = (tag: ts.JSDocTag) => {
+  let {comment} = tag;
+  if (comment === undefined) {
+    return undefined;
+  }
+  if (Array.isArray(comment)) {
+    comment = comment.map((c) => c.text).join('');
+  }
+  if (typeof comment !== 'string') {
+    throw new DiagnosticsError(tag, `Internal error: unsupported node type`);
```

Comment:
Again unsure if we want to throw here... let's chat about error handling.

Reply:
Same as above, lets handle systemically per https://github.com/lit/lit/issues/3009